assets/wizards/popups/components/settings-modal/index.js

```diff
@@ -191,7 +198,18 @@ const PromptSettingsModal = ( { prompt, disabled, onClose, segments, updatePopup
 								hideHelpFromVision
 								taxonomy="tags"
 								value={ excludedTags || [] }
-								onChange={ tokens => setPromptConfig( { options: { excluded_tags: tokens } } ) }
+								onChange={ tokens => {
+									return setPromptConfig( {
+										options: {
+											excluded_tags: tokens.reduce( ( acc, token ) => {
+												if ( token?.id ) {
```

Comment:
Why shouldn't the `CategoryAutocomplete` component do the filtering?

Reply:
The `onChange` function is passed as a prop to `CategoryAutocomplete`. I tried adding some filtering to the `handleOnChange` helper in the component, but it doesn't prevent the `undefined` error so the filtering is still required in the function passed as `onChange`.

That said, cc662e7 does handle the filtering in a slightly more readable way, and I retained the extra filtering in `CategoryAutocomplete` just in case.

Reply:
If the filtering happens after the mapping in `handleOnChange`, the value passed to the `onChange` will always be valid (there will be no need for filtering in `onChange`):

```js
const allValues = tokens
	.map( token => ( 'string' === typeof token ? suggestions[ token ] : token ) )
	.filter( Boolean );
```

Reply:
Ah, you're right. Updated in 94a399e. I was putting `filter` before the `map`, which was still causing the `undefined` error.